src/test/java/com/olgaboiar/mint/RequestBuilderTest.java

```diff
@@ -0,0 +1,35 @@
+package com.olgaboiar.mint;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RequestBuilderTest {
+    RequestBuilder testRequestBuilder;
+
+    @BeforeEach
+    public void init() throws IOException {
+        MockServerConnection testSocket = new MockServerConnection();
+        BufferedReader in = testSocket.listenToClientConnection(testSocket.acceptClientConnection());
+        testRequestBuilder = new RequestBuilder(new RequestParser(), new Reader(in));
+    }
+
+    @Test
+    void testRequestBuilderCreatesRequestObject() throws IOException {
+        List<String> incomingRequest = new ArrayList<String>();
+        incomingRequest.add("HEAD /simple_get HTTP/1.1");
+        incomingRequest.add("Accept: */*");
+        incomingRequest.add("User-Agent: Ruby");
+        incomingRequest.add("Connection: close");
+        incomingRequest.add("Host: 0.0.0.0:5000");
+        Request parsedRequest = testRequestBuilder.buildRequest(incomingRequest);
+
+        assertTrue(parsedRequest instanceof Request);
```

Comment:
Perhaps we can check the content of the request instead of its type? Since parseRequest is of type Request which you declared, this assertion isn't testing what you're trying to test.

Reply:
refactored in https://github.com/olgaboiar/MintHTTPServer/pull/10/commits/95ac3d5170fd3380362edf2f4a279d0caf9d9b94